site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -137,7 +137,22 @@ public function deleteCourseMaterial($id) {
         else {
             $success = unlink($path);
         }
-
+        $base_path = $this->core->getConfig()->getCoursePath() . "/uploads/course_materials";
+        // delete the topmost parent folder that's empty (contains no files)
+        if (pathinfo($path, PATHINFO_DIRNAME) !== $base_path) {
+            $empty_folders = [];
+            FileUtils::getTopEmptyDir($path, $base_path, $empty_folders);
+            if (count($empty_folders) > 0) {
+                $path = $empty_folders[0];
+                $success = $success && FileUtils::recursiveRmdir($path);
+                foreach ($all_files as $file) {
+                    if (str_starts_with($file->getPath(), $path)) {
+                        $this->core->getCourseEntityManager()->remove($file);
+                    }
+                }
+                $this->core->getCourseEntityManager()->flush();
+            }
```

Comment:
Why remove all files again? Doesn't this get done in 127-131?

Reply:
`$path` here is different (equal to `$empty_folders[0]`), this hasn't been removed previously.

Reply:
I see.  Thanks for clarification. Couldn't this be simplified to just remove the single top directory? We know it's empty so there shouldn't be any course materials under it.

Reply:
The definition of 'empty' here is having no files, I think it could contain empty subfolders inside it, so, to remove them iteration is necessary.